tls/s2n_connection.c

```diff
@@ -909,11 +910,28 @@ int s2n_connection_set_read_fd(struct s2n_connection *conn, int rfd)
     return 0;
 }
 
+int s2n_connection_get_read_fd(struct s2n_connection *conn, int *readFd)
+{
+    POSIX_ENSURE_REF(conn);
+    POSIX_ENSURE_REF(readFd);
+
+    /* in case we do not use managed io, or in case read fd was never set we should return invalid fd */
+    if(conn->managed_io == 1 && conn->recv_io_context) {
+        struct s2n_socket_read_io_context *peer_socket_ctx = conn->recv_io_context;
+        *readFd = peer_socket_ctx->fd;
+    }
+    else {
+        *readFd = -1;
+    }
+    return S2N_SUCCESS;
```

Comment:
Should we be returning an error instead if someone tries to retrieve a fd when there isn't a fd set?

Reply:
I would agree with this. I would remove the `*readfd = -1` and do a `POSIX_BAIL` instead.

Reply:
I have no major objection for removing it but
I think that in case the fd is not set it is NOT an error.
this way enables the user to distinguish case were internal error or bad state detected and the case were the user only set the read/write side in managed IO